src/table/secrets_settings.ini

```diff
@@ -0,0 +1,84 @@
+; This file is part of OpenTTD.
+; OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+; OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+; See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+;
+
+[pre-amble]
+static bool UpdateServerPassword(int32 p1);
+static bool UpdateRconPassword(int32 p1);
+
+static const SettingTable _secrets_settings = {
+[post-amble]
+};
+[templates]
+SDTC_BOOL  =  SDTC_BOOL(       $var,        $flags, $guiflags, $def,                              $str, $strhelp, $strval, $proc, $from, $to,        $cat, $extra, $startup),
```

Comment:
If secrets cannot be passed over the network or saved, maybe replace `$flags` with `$flags | SLF_NOT_IN_SAVE | SLF_NO_NETWORK_SYNC` and `$gui_flags` with `$guiflags | SGF_NETWORK_ONLY` here. Then if someone wants to do it, they need to change the templates which would be super visible in the PR. Furthermore it removes the need to specify it on all the settings separately.

Reply:
I have made some drafts for this, but they didn't make it into the PR. Mostly as I first want to know what we want to put in `secrets.cfg`. If we settled on that, we can indeed make this all a bit smarter :D (similar for why there are additional template entries here atm).